src/channel/ably/auth.ts

```diff
@@ -0,0 +1,139 @@
+import { beforeChannelAttach } from './attach';
+import { httpReqFunction, parseJwt, toTokenDetails } from './utils';
+import { SequentialAuthTokenRequestExecuter } from './token-request';
+import { AblyChannel } from '../ably-channel';
+import { AblyConnector } from '../../connector/ably-connector';
+import { AblyPresenceChannel } from '../ably-presence-channel';
+
+export class AblyAuth {
+
+    // TODO - Can be updated with request throttle, to send multiple request payload under single request
+    authRequestExecuter: SequentialAuthTokenRequestExecuter;
+    authEndpoint = '/broadcasting/auth';
+    authHost = typeof window != 'undefined' && window?.location?.hostname;
+    authPort = typeof window != 'undefined' && window?.location?.port;
+    authProtocol = typeof window != 'undefined' && window?.location?.protocol.replace(':', '');
+
+    httpReq = httpReqFunction();
+
+    authOptions = {
+        queryTime: true,
+        useTokenAuth: true,
+        authCallback: async (_, callback) => { // get token from tokenParams
+            try {
+                const { token } = await this.authRequestExecuter.request(null); // Replace this by network request to PHP server
```

Comment:
I'm not sure what this comment means, can you clarify?

Reply:
Actually comment is not needed, I will remove it : )

Reply:
Updated!